diff --git a/CMakeLists.txt b/CMakeLists.txt
index fab4e89..dfea35d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,33 +67,13 @@ endif()
 # Vulkan transitive dependency
 
 if(VOLK_PULL_IN_VULKAN)
-  # Try an explicit CMake variable first, then any Vulkan paths
-  # discovered by FindVulkan.cmake, then the $VULKAN_SDK environment
-  # variable if nothing else works.
-  if(VULKAN_HEADERS_INSTALL_DIR)
-    message("volk: using VULKAN_HEADERS_INSTALL_DIR option")
-    set(VOLK_INCLUDES "${VULKAN_HEADERS_INSTALL_DIR}/include")
-  else()
-    # If CMake has the FindVulkan module and it works, use it.
-    find_package(Vulkan QUIET)
-    if(Vulkan_INCLUDE_DIRS)
-      message("volk: using Vulkan_INCLUDE_DIRS from FindVulkan module")
-      set(VOLK_INCLUDES "${Vulkan_INCLUDE_DIRS}")
-    elseif(DEFINED ENV{VULKAN_SDK})
-      message("volk: using VULKAN_SDK environment variable")
-      set(VOLK_INCLUDES "$ENV{VULKAN_SDK}/include")
-    elseif(TARGET Vulkan-Headers)
-      message("volk: using Vulkan-Headers include directories")
-      get_target_property(VOLK_INCLUDES Vulkan-Headers INTERFACE_INCLUDE_DIRECTORIES)
-    endif()
-  endif()
+  find_package(VulkanHeaders CONFIG REQUIRED)
 
-  if(VOLK_INCLUDES)
-    if(TARGET volk)
-      target_include_directories(volk PUBLIC "${VOLK_INCLUDES}")
-    endif()
-    target_include_directories(volk_headers INTERFACE "${VOLK_INCLUDES}")
+  if(TARGET volk)
+    target_link_libraries(volk PUBLIC Vulkan::Headers)
   endif()
+
+  target_link_libraries(volk_headers INTERFACE Vulkan::Headers)
 endif()
 
 # -----------------------------------------------------
