diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0fed973bd..28be5feaa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -664,28 +664,26 @@ configure_package_config_file(
 # linkable targets. In this case do not export the targets. Otherwise, just
 # export the slang targets.
 if(NOT CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
-    if(NOT ${SLANG_LIB_TYPE} STREQUAL "STATIC")
-        install(
-            EXPORT SlangTargets
-            FILE ${PROJECT_NAME}Targets.cmake
-            NAMESPACE ${PROJECT_NAME}::
-            DESTINATION ${SLANG_CMAKE_CONFIG_DIR}
-        )
+    install(
+        EXPORT SlangTargets
+        FILE ${PROJECT_NAME}Targets.cmake
+        NAMESPACE ${PROJECT_NAME}::
+        DESTINATION ${SLANG_CMAKE_CONFIG_DIR}
+    )
 
-        if(UNIX)
-            configure_file(
-                "${PROJECT_SOURCE_DIR}/extras/pkgconfig/slang-compiler.pc.in"
-                "${PROJECT_BINARY_DIR}/slang-compiler.pc"
-                @ONLY
-            )
+    if(UNIX)
+        configure_file(
+            "${PROJECT_SOURCE_DIR}/extras/pkgconfig/slang-compiler.pc.in"
+            "${PROJECT_BINARY_DIR}/slang-compiler.pc"
+            @ONLY
+        )
 
-            # TODO: When build system is changed to respect CMAKE_INSTALL_LIBDIR,
-            # update this destination.
-            install(
-                FILES "${PROJECT_BINARY_DIR}/slang-compiler.pc"
-                DESTINATION "lib/pkgconfig"
-            )
-        endif()
+        # TODO: When build system is changed to respect CMAKE_INSTALL_LIBDIR,
+        # update this destination.
+        install(
+            FILES "${PROJECT_BINARY_DIR}/slang-compiler.pc"
+            DESTINATION "lib/pkgconfig"
+        )
     endif()
 endif()
 
