diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2b9ef01..a41631b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,7 +47,7 @@ elseif (TARGET Vulkan-HppModule)
     set(VMA_HPP_VULKAN_MODULE Vulkan-HppModule)
 endif ()
 
-add_subdirectory(VulkanMemoryAllocator)
+find_package(VulkanMemoryAllocator REQUIRED)
 add_subdirectory(include)
 
 if (NOT ${PROJECT_VERSION} STREQUAL ${VMA_HPP_PROJECT_VERSION})
diff --git a/include/CMakeLists.txt b/include/CMakeLists.txt
index 037035d..7aeddc3 100644
--- a/include/CMakeLists.txt
+++ b/include/CMakeLists.txt
@@ -13,6 +13,7 @@ add_library(VulkanMemoryAllocator-Hpp INTERFACE)
 add_library(VulkanMemoryAllocator-Hpp::VulkanMemoryAllocator-Hpp ALIAS VulkanMemoryAllocator-Hpp)
 add_library(VulkanMemoryAllocator::Hpp ALIAS VulkanMemoryAllocator-Hpp)
 target_include_directories(VulkanMemoryAllocator-Hpp INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
+target_link_libraries(VulkanMemoryAllocator-Hpp INTERFACE GPUOpen::VulkanMemoryAllocator)
 
 if (TARGET Vulkan::HppModule)
     message(STATUS "VMA-Hpp module is available")
